pkg/controller/mapper.go

```diff
@@ -75,3 +79,53 @@ func TypeMapperWithinNamespace(cl client.Client, extensionType string) handler.T
 		return nil
 	}
 }
+
+type clusterToObjectMapper struct {
+	client         client.Client
+	newObjListFunc func() runtime.Object
+	predicates     []predicate.Predicate
+}
+
+func (m *clusterToObjectMapper) Map(obj handler.MapObject) []reconcile.Request {
+	if obj.Object == nil {
+		return nil
+	}
+
+	cluster, ok := obj.Object.(*extensionsv1alpha1.Cluster)
+	if !ok {
+		return nil
+	}
+
+	objList := m.newObjListFunc()
+	if err := m.client.List(context.TODO(), client.InNamespace(cluster.Name), objList); err != nil {
+		return nil
+	}
+
+	var requests []reconcile.Request
+	_ = meta.EachListItem(objList, func(obj runtime.Object) error {
```

Comment:
How do you suggest to handle it? What else can I do other than not adding anything to the `[]reconcile.Request` list (which is the same like now)?

Reply:
I'd probably either do
* `utilruntime.HandleError` (handles user facing errors)
* Return without adding any request because this means that the list is malformed / has malformed entries which is rather a design time issue.